broker-daemon/state-machines/plugins/state-machine-dates.js

```diff
@@ -0,0 +1,36 @@
+const StateMachinePlugin = require('./abstract')
+
+/**
+ * @class Store dates when states changed
+ */
+class StateMachineDates extends StateMachinePlugin {
+  /**
+   * Set up configuration for the dates plugin, controlling which properties on the host object to use
+   * @param {Object} options
+   * @param {Array<string>} [options.skipTransitions=[]] - Array of transition names to be ignored
+   */
+  constructor ({ skipTransitions = [] } = {}) {
+    super()
+    this.skipTransitions = skipTransitions
+  }
+
+  /**
+   * Observers object to add additional lifecycle observers
+   * Adds our lifecycle observers to add a date to StateMachine when a new state is entered
+   * @returns {Object} Key value of observers
+   */
+  get observers () {
+    const plugin = this
+
+    return {
+      onEnterState: function (lifecycle) {
+        if (!plugin.skipTransitions.includes(lifecycle.transition)) {
+          // set the date the state was entered, for example, 'dateCreated'
+          this.dates[`date${this.state.charAt(0).toUpperCase()}${this.state.slice(1)}`] = new Date()
```

Comment:
since this is already in a property called `dates`, can't we just call this `created`?

e.g. this.dates['created'] = blah

That seems clear enough to me and avoids any string manipulation

Reply:
Yeah i think you're right. I had this before we had that dates object.